### PR TITLE
Generic post-processing task for quality objects

### PR DESCRIPTION
**Pull request by aferrero2707**
The task provides trending of user-defined quality flags as well as a summary in human-readable format in a dedicated canvas. An example of configuration is provided in postprocessing.json

### CODE REVIEW

**Comment by Barthelemy** [comment]
thank you !

**Comment by aferrero2707** [comment]
looks like I need to rebase due to recent changes in the post-processing interface, will do it asap.

**Comment by aferrero2707** [comment]
@Barthelemy @knopers8 I have added a section for the QualityTask in the post-processing documentation.
Two questions:
* should the code be put in `Framework` (like for the `TrendingTask`) or in `Modules/Common` (like for the `TRFCollectionTask`)?
* do you prefer to list the input QOs via `dataSources`, or a dedicated `QOs` array like in `TRFCollectionTask`?

**Comment by Barthelemy** [comment]
@aferrero2707 I will let @knopers8 answer

File: Framework/src/QualityTask.cxx [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+//_________________________________________________________________________________________
+// Helper function for retrieving a MonitoringObject from the QCDB.
+// A non-null MO is returned if:
+// - the time stamp of the object is newer than the last visited one
+// - the MO contains a QualityObject
+
+static std::shared_ptr<MonitorObject> getMO(repository::DatabaseInterface& qcdb, Trigger t, QualityTaskConfig::DataSource& source)
+{
+  // retrieve MO from CCDB
+  auto mo = qcdb.retrieveMO(source.path, source.name, t.timestamp, t.activity);
```
[the remark this answers is not in the document]
Comment (aferrero2707) [reply Framework/src/QualityTask.cxx]: Done

File: Framework/src/QualityTask.cxx [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+    // retrieve MO from CCDB
+    // a valid object is returned only if the QCDB entry has a time stamp more recent than the last retrieved one
+    auto mo = getMO(qcdb, t, source);
+    if (!mo) {
+      continue;
+    }
+    // get corresponding QO
+    auto qo = dynamic_cast<QualityObject*>(mo->getObject());
+
+    auto time = source.timeStamp / 1000; // ROOT expects seconds since epoch
```
[the remark this answers is not in the document]
Comment (aferrero2707) [reply Framework/src/QualityTask.cxx]: Done.

File: Framework/src/QualityTask.cxx [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+
+    std::string Qname = qo->getName();
+    std::string Qvalue = qo->getQuality().getName();
+    int Q = mQualityIDs[Qvalue];
+    std::string Qstr = fmt::format("#color[{}]", mColors[Qvalue]) + "{" + Qvalue + "}";
```
[the remark this answers is not in the document]
Comment (aferrero2707) [reply Framework/src/QualityTask.cxx]: Ok, I have changed the variables as qoXXX.

File: Framework/src/QualityTask.cxx [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+  msg->SetBorderSize(0);
+  msg->SetFillColor(kWhite);
+  msg->Draw();
+}
+
+//_________________________________________________________________________________________
+
+void QualityTask::finalize(Trigger t, framework::ServiceRegistryRef)
+{
+}
```
[the remark this answers is not in the document]
Comment (aferrero2707) [reply Framework/src/QualityTask.cxx:257]: We should discuss what is the best strategy here. I see that TPC in the QualityObserver re-generates the canvas but does not retrieve the QOs.

**Comment by aferrero2707** [comment]
@knopers8 code moved into `Modules/Common` and requested changes implemented.

**Comment by knopers8** [comment]
@aferrero2707 So do we merge it or you would still like to test it?

**Comment by aferrero2707** [comment]
@knopers8 sorry, had no time to play with it further, but I would propose to merge and then test it on staging first, and in production if all goes well. What do you think?

**Comment by knopers8** [comment]
OK, fine with me!

**Review by knopers8 — APPROVED** [review]

**Pull request merged**